compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -154,9 +154,9 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
   lazy val Predef_classOf: Symbol = defn.ScalaPredefModule.requiredMethod(nme.classOf)
```

Comment:
That doesn't work as moving it the companion of PreName is not enough to make `.toTermName` and `.toTypeName` available on Strings. I guess we could have two implicit conversions, one in PreName's companion and one available by import, but I'm feel it would just make things worse...

Are you sure you don't like this commit as is? In my opinion completely removing an abstraction from the code base is a clear simplification. The diff might look large but in reality most of the line changes are about making type *more precise* by having the `TermName`/`TypeName` instead of `PreName`. The balance sheet in term of added boilerplate is also reasonable:

`toTermName` added: 22
`toTermName` removed: 12

`toTypeName` added: 12
`toTypeName` removed: 4

+4 overloaded methods with String & Name arguments

Reply:
I still think we should leave PreName. I don't like to have to overload methods, and the problem is, if you are systematic about it (and you should be!) you will end up with a lot more overloaded methods. `PreName` is a single abstraction which fixes this quite nicely. For me it is really analogous to `IterableOnce` in collections. Before we had some overloaded methods that took `Iterable`s and `Iterator`s respectively. At first there were only a few. Then people complained that, by rights, some other methods should also be dual purpose, and then some others more, and so on. The introduction of `IterableOnce` stopped the bleeding. 

I realize it's a contentious issue, but then the maxime "when in doubt, don't change it" applies.

One other thing: We have to tell everyone working on the compiler that `Decorators` is a highly recommended (maybe mandatory?) input. I have seen so much awkward code because people did not know about it. Is there a good place where we can put it so people will read it?